module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -322,89 +309,165 @@ namespace vision {
+                                          "maximum_relative_seed_point_distance");
+
+                                      log("arma::max(sDist) = ",
+                                          arma::max(sDist),
+                                          " > ",
+                                          maximum_relative_seed_point_distance);
+                                  }
+                                  continue;
+                              }
+
+                              // BALL IS CLOSER THAN 1/2 THE HEIGHT OF THE ROBOT BY WIDTH
+                              double widthDistance = widthBasedDistanceToCircle(field.ball_radius, top, base, cam);
+
+                              if (widthDistance < cameraHeight * 0.5) {
+                                  if (print_throwout_logs) {
+                                      log("Ball discarded: widthDistance < cameraHeight * 0.5");
+                                  }
+                                  continue;
+                              }
+
+                              // IF THE DISAGREEMENT BETWEEN THE WIDTH AND PROJECTION BASED DISTANCES ARE TOO LARGE
+                              // Project this vector to a plane midway through the ball
+                              Plane ballBisectorPlane({0, 0, 1}, {0, 0, field.ball_radius});
+                              arma::vec3 ballCentreGroundProj =
+                                  projectCamToPlane(ballCentreRay, camToGround, ballBisectorPlane);
+                              double ballCentreGroundProjDistance = arma::norm(ballCentreGroundProj);
+
+                              if (std::abs((widthDistance - ballCentreGroundProjDistance)
+                                           / std::max(ballCentreGroundProjDistance, widthDistance))
+                                  > MAXIMUM_DISAGREEMENT_RATIO) {
+                                  if (print_throwout_logs) {
+                                      log("Ball discarded: Width and proj distance disagree too much: width =",
+                                          widthDistance,
+                                          "proj =",
+                                          ballCentreGroundProjDistance);
+                                  }
+                                  continue;
+                              }
+
+                              /************************************************
+                               *                 MEASUREMENTS                 *
+                               ************************************************/
+
+                              Ball b;
+                              b.visObject.sensors = image.sensors;
+
+                              // Work out how far away the ball must be to be at the distance it is from the camera
+                              arma::vec3 width_rBCc = ballCentreRay * widthDistance;
+
+                              arma::vec3 rBCc = (width_rBCc);
```

Comment:
Why is this here? Why not just `arma::vec3 rBCc = ballCentreRay * widthDistance'`?

Reply:
I believe previously it was the average of the two measurements, now it seems it's just one